### PR TITLE
ci: remove cargo registry cache

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

With sparse-registry and sccache, I think we don't need to cache registry anymore.

## Checklist For Contributors

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [ ] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by xxchan** [comment]
We don't need to cache registry anymore 🤡

**Comment by xxchan** [comment]
Why build has this 44s? 
- It's not cache, as deleted in this PR
- It's not sccache. Adding a dummy echo can confirm.
- Is it docker-compose pre exit hook?

P.S. build-simulation doesn't have this.

image

**Comment by xxchan** [comment]
Wow, success. Why `rm -rf target` can remove the 44s? 😅

**Comment by BugenZhao** [comment]
> Wow, success. Why `rm -rf target` can remove the 44s? 😅

Have we found out what the magic is? 👀

**Comment by xxchan** [comment]
> > Wow, success. Why `rm -rf target` can remove the 44s? 😅
> 
> Have we found out what the magic is? 👀

I have no idea. Just guess it's `docker rm` being slow. 

There are some articles like mentioning `devicemapper` is slow (for Arch Linux). https://medium.com/@shagabutdinov/fixing-slow-docker-rm-d70f98024fb3

We are using Amazon Linux and 
```
Storage Driver: overlay2
--
  | Backing Filesystem: xfs
  | Supports d_type: true
  | Native Overlay Diff: true
  | userxattr: false
```

No much knowledge about it.

File: ci/workflows/pull-request.yml [hunk 1]
```diff
@@ -227,7 +197,18 @@ steps:
   - label: "check"
     command: "ci/scripts/check.sh"
     plugins:
-      - gencer/cache#v2.4.10: *cargo-cache
+      - gencer/cache#v2.4.10:
```
Comment (BugenZhao) [inline ci/workflows/pull-request.yml:200]: Any reason for still keeping this?

Comment (xxchan) [reply ci/workflows/pull-request.yml:200]: 59s vs 4s 🤔 
image

**Pull request merged**